src/main/scala/it/pagopa/interop/authorizationmanagement/model/persistence/State.scala

```diff
@@ -11,6 +11,20 @@ final case class State(keys: Map[ClientId, Keys], clients: Map[ClientId, Persist
     case None          => this
   }
 
+  def updateKey(event: KeyUpdated): State = keys.get(event.clientId) match {
+    case Some(entries) => {
+      entries.get(event.keyId) match {
+        case Some(key) => {
+          val updatedKey = key.copy(userId = Some(event.userId))
+          val filtered   = entries.filterNot(_._1 == updatedKey.kid)
+          copy(keys = keys + (event.clientId -> (filtered + (updatedKey.kid -> updatedKey))))
```

Comment:
if `+` on `Map` replaces the existing key, is this equivalent to
```suggestion
          val updatedKey = key.copy(userId = Some(event.userId))
          copy(keys = keys + (event.clientId -> (entries + (updatedKey.kid -> updatedKey))))
```
?